output/statsd/output.go

```diff
@@ -0,0 +1,188 @@
+/*
+ *
+ * k6 - a next-generation load testing tool
+ * Copyright (C) 2019 Load Impact
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package statsd
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/DataDog/datadog-go/statsd"
+	"github.com/sirupsen/logrus"
+
+	"github.com/loadimpact/k6/output"
+	"github.com/loadimpact/k6/stats"
+)
+
+// New creates a new statsd connector client
+func New(params output.Params) (output.Output, error) {
+	return newOutput(params)
+}
+
+func newOutput(params output.Params) (*Output, error) {
+	conf, err := getConsolidatedConfig(params.JSONConfig, params.Environment, params.ConfigArgument)
+	if err != nil {
+		return nil, err
+	}
+	logger := params.Logger.WithFields(logrus.Fields{"output": "statsd"})
+
+	return &Output{
+		config: conf,
+		logger: logger,
+	}, nil
+}
+
+var _ output.Output = &Output{}
+
+// Output sends result data to statsd daemons with the ability to send to datadog as well
+type Output struct {
+	output.SampleBuffer
+
+	periodicFlusher *output.PeriodicFlusher
+
+	config config
+
+	logger logrus.FieldLogger
+	client *statsd.Client
+}
+
+func (o *Output) dispatch(entry stats.Sample) error {
+	var tagList []string
+	switch o.config.TagFormat.t {
+	case tagFormatDatadog, tagFormatNewrelic, tagFormatStatsdv2:
+		tagList = processTags(o.config.TagBlocklist, entry.Tags.CloneTags())
```

Comment:
I kind of see @na--'s comment about this: if they all use the same format, why not just drop this or leave only `tagFormatStatsdv2`?

Though I guess it would be useful to have to handle any inconsistencies and eventually support the graphite format as well.

Reply:
statsd prior to v0.9.0 *does* not support the tag format so if someone is still using it and updates k6 it will stop working for them. Also arguably `graphite` and possibly other pieces of software that support the non-tag statsd format but not the "datadog" one will also possibly *stop* working.

So IMO we need at least `none` and one more whether we have 3 names for the same thing makes very little difference in the actual code (as you can see), so I prefer to have less confusion and just name them accordingly.

Reply:
statsd prior to v0.9.0 *does* not support the tag format so if someone is still using it and updates k6 it will stop working for them. Also arguably `graphite` and possibly other pieces of software that support the non-tag statsd format but not the "datadog" one will also possibly *stop* working.

So IMO we need at least `none` and one more whether we have 3 names for the same thing makes very little difference in the actual code (as you can see), so I prefer to have less confusion and just name them accordingly.

Reply:
I guess I don't see the point in exposing different vendor formats if they're all essentially the same and could be grouped under a generic `statsdv2` format. Seems cosmetic for no good reason. Similar to what [@simskij mentioned](https://github.com/k6io/k6/pull/1963#issuecomment-821144078).